src/components/InputPassword.js

```diff
@@ -83,10 +87,6 @@ class InputPassword extends Component {
     password: ""
   };
 
-  componentDidMount() {
```

Comment:
Are you sure they need to be removed? i.e. Component should return ref for input when it mounts so parent application has access to it immediately.

Reply:
No, okay so whats happening is `<InputPassword />` extends `<Input />` which will handle the getRef as follows ...
<img width="266" alt="screen shot 2017-10-04 at 10 38 32" src="https://user-images.githubusercontent.com/6928300/31169348-34be8ef6-a8f0-11e7-9771-8d25c6889aa6.png">

<img width="427" alt="screen shot 2017-10-04 at 10 36 34" src="https://user-images.githubusercontent.com/6928300/31169296-0cb3dea2-a8f0-11e7-816c-79cb338a1d25.png">

`getRef` is handled by `<Input />` so it is available as soon as the component mounts, which is why components which extend it don't need to implement the same logic, the prop is hoisted.

Reply:
Any component which doesn't implement it i.e `<InputIncrement />` will implement the logic in there scope...
<img width="724" alt="screen shot 2017-10-04 at 10 43 04" src="https://user-images.githubusercontent.com/6928300/31169562-f125dc5c-a8f0-11e7-8b13-ed4f06c5a242.png">